Haskintex.hs

```diff
@@ -295,22 +303,36 @@ memoreduce modName isMemo t ty f = do
       -- Sandbox recognition
       inSandbox <- lift $ doesDirectoryExist ".cabal-sandbox"
       r <- if inSandbox
-              then do outputStr "Sandbox detected."
-                      noSandbox <- nosandboxFlag <$> get
-                      if noSandbox
-                         then do outputStr "Ignoring sandbox."
-                                 runInterpreter int
-                         else do sand <- lift $ getDirectoryContents ".cabal-sandbox"
-                                 let pkgdbs = filter (isSuffixOf "packages.conf.d") sand
-                                 case pkgdbs of
-                                   pkgdb : _ -> do
-                                     outputStr $ "Using sandbox package db: " ++ pkgdb
-                                     unsafeRunInterpreterWithArgs ["-package-db .cabal-sandbox/" ++ pkgdb] int
-                                   _ -> runInterpreter int
+              then do
+                outputStr "Sandbox detected."
+                noSandbox <- nosandboxFlag <$> get
+                if noSandbox
+                  then do
+                    outputStr "Ignoring sandbox."
+                    runInterpreter int
+                  else do
+                    stackDb <- stackDbFlag <$> get
+                    if stackDb
+                      then do
+                        pkgdb <- shelly $ bash "stack path" ["--local-pkg-db"]
+                        outputStr $ "Using sandbox package db: " ++ unpack pkgdb
+                        unsafeRunInterpreterWithArgs ["-package-db " ++ unpack pkgdb, "-package yaml"] int
+                      else do
+                        sand <- lift $ getDirectoryContents ".cabal-sandbox"
+                        let pkgdbs = filter (isSuffixOf "packages.conf.d") sand
+                        case pkgdbs of
+                          pkgdb : _ -> do
+                            outputStr $ "Using sandbox package db: " ++ pkgdb
+                            unsafeRunInterpreterWithArgs ["-package-db .cabal-sandbox/" ++ pkgdb] int
+                          _ -> runInterpreter int
```

Comment:
Sorry, I took months to review this...

First of all, I recommend not making layout changes when making PRs (unless it's a PR about layout changes!). This is because it forces the maintainer to look carefully and compare more lines than what it's actually needed. In this particular PR is not a big deal, it's only a few lines. But it's just a recommendation. :)

Did you add the shelly dependency for a reason? What you do there can be done as easily with the process package, a package haskintex already depends on.

The last question is probably out of my ignorance, but why the `-package yaml` flag when running the interpreter?

Thanks for your contribution and sorry for the late review!